src/generic/datavgen.cpp

```diff
@@ -3277,7 +3275,7 @@ void wxDataViewMainWindow::SendSelectionChangedEvent( const wxDataViewItem& item
 
 void wxDataViewMainWindow::RefreshRow( unsigned int row )
 {
-    wxRect rect( 0, GetLineStart( row ), GetEndOfLastCol(), GetLineHeight( row ) );
+    wxRect rect( 0, GetLineStart( row ), INT_MAX, GetLineHeight( row ) );
```

Comment:
I think it might be worth it to create `GetPhysicalLinesRect(from, to)` containing this and the next line if only to write a comment explaining why do we use `INT_MAX` here and not `GetEndOfLastCol()` once, and use this function from both here and below.

Reply:
I agree — but doing it led me on a bit of a goose chase, as there's much shared by the two methods… all of them, basically. So I now pushed a version that has `RefreshRow(row)` call `RefreshRows(row,row)` and kept scrolling logic in it.